controllers/lease_controller.go

```diff
@@ -0,0 +1,456 @@
+	logger.Info("Reconcile etcd compaction job")
+
+	// First check if a job is already running
+	job := &batchv1.Job{}
+	err := lc.Get(ctx, types.NamespacedName{Name: getJobName(etcd), Namespace: etcd.Namespace}, job)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Required job doesn't exist. Create new
+			job, err = lc.CreateCompactJob(ctx, logger, etcd)
+			if err != nil {
+				logger.Error(err, "error during compaction job creation")
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	// Delete job and reque if the job failed
+	if job.Status.Failed > 0 {
+		err = lc.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground))
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("job status failed")
+	}
+
+	// Delete job and return if the job succeeded
+	if job.Status.Succeeded > 0 {
+		err = lc.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground))
+		if err != nil {
+			return nil, err
+		}
+		return job, nil
+	}
+
+	// Delete job and reque if the job doesn't have label with this etcd instance
+	if val, ok := job.Labels["instance"]; ok {
+		if val != etcd.Name {
+			err = lc.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground))
+			if err != nil {
+				return nil, err
+			}
+			return nil, fmt.Errorf("job doesn't have label for this etcd instance")
+		}
+	}
+
+	return job, nil
+}
+
+func (lc *LeaseController) CreateCompactJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (*batchv1.Job, error) {
+	/*resourceValues := backupValues["compactionResources"].(map[string]interface{})
+	requestValues := resourceValues["requests"].(map[string]interface{})
+	limitValues := resourceValues["limits"].(map[string]interface{})*/
+
+	activeDeadlineSeconds, err := time.ParseDuration(lc.config.ActiveDeadlineDuration)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse activeDeadlineDuration string from custodian config: %v", err)
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getJobName(etcd),
+			Namespace: etcd.Namespace,
+			Labels:    getLabels(etcd),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "druid.gardener.cloud/v1alpha1",
+					BlockOwnerDeletion: pointer.BoolPtr(true),
+					Controller:         pointer.BoolPtr(true),
+					Kind:               "Etcd",
+					Name:               etcd.Name,
+					UID:                etcd.UID,
+				},
+			},
+		},
+
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds: pointer.Int64Ptr(int64(activeDeadlineSeconds.Seconds())),
+			BackoffLimit:          pointer.Int32Ptr(0),
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: etcd.Spec.Annotations,
+					Labels:      getLabels(etcd),
+				},
+				Spec: v1.PodSpec{
+					//ServiceAccountName: fmt.Sprint(values["serviceAccountName"]),
```

Comment:
Needs to be uncommented before merging to add a serviceaccount to the compaction job
This is needed for backup sidecar to be able to update the snapshot leases